src/covid_model_seiir_pipeline/postprocessing/data.py

```diff
@@ -0,0 +1,291 @@
+        covariates = covariates.set_index(['location_id', 'date']).sort_index()
+        if time_varying:
+            covariate = covariates[covariate].rename(draw_id)
+        else:
+            covariate = covariates.groupby(level='location_id')[covariate].max().rename(draw_id)
+        return covariate
+
+    def load_input_covariate(self, covariate: str, covariate_version: str, location_ids: List[int]):
+        return self._get_forecast_data_inteface().load_covariate(covariate, covariate_version,
+                                                                 location_ids, with_observed=True)
+
+    def load_betas(self, draw_id: int, scenario: str) -> pd.Series:
+        ode_params = io.load(self.forecast_root.ode_params(scenario=scenario, draw_id=draw_id))
+        ode_params['date'] = pd.to_datetime(ode_params['date'])
+        betas = (ode_params
+                 .set_index(['location_id', 'date'])['beta']
+                 .sort_index()
+                 .rename(draw_id))
+        return betas
+
+    def load_beta_residuals(self, draw_id: int) -> pd.Series:
+        beta_regression = self._get_forecast_data_inteface().load_beta_regression(draw_id)
+        beta_regression = (beta_regression
+                           .set_index(['location_id', 'date'])
+                           .sort_index()[['beta', 'beta_pred']])
+        beta_residual = np.log(beta_regression['beta'] / beta_regression['beta_pred']).rename(draw_id)
+        return beta_residual
+
+    def load_raw_outputs(self, draw_id: int, scenario: str) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+        draw_df = io.load(self.forecast_root.raw_outputs(scenario=scenario, draw_id=draw_id))
+        draw_df = draw_df.set_index(['location_id', 'date']).sort_index()
+        deaths = draw_df.reset_index().set_index(['location_id', 'date', 'observed'])['deaths'].rename(draw_id)
+        infections = draw_df['infections'].rename(draw_id)
+        r_effective = draw_df['r_effective'].rename(draw_id)
+        return deaths, infections, r_effective
+
+    def load_elastispliner_inputs(self) -> pd.DataFrame:
+        metadata = self._get_forecast_data_inteface().get_infectionator_metadata()
+        deaths_version = Path(metadata['death']['metadata']['output_path'])
+        es_inputs = pd.read_csv(deaths_version / 'model_data.csv')
+        es_inputs['date'] = pd.to_datetime(es_inputs['date'])
+        es_inputs = es_inputs.set_index(['location_id', 'date'])
+        cumulative_cases = (es_inputs['Confirmed case rate'] * es_inputs['population']).rename('cumulative_cases')
+        cumulative_deaths = (es_inputs['Death rate'] * es_inputs['population']).rename('cumulative_deaths')
+        cumulative_hospitalizations = (es_inputs['Hospitalization rate'] * es_inputs['population'])
+        cumulative_hospitalizations = cumulative_hospitalizations.rename('cumulative_hospitalizations')
+        # TODO: have ES use and produce a config file for this number.
+        es_shift = -8
```

Comment:
This is new. The full_data csv containing input data doesn't fill missing days, but that makes big trouble for the viz.  We want stuff after ES has filled gaps, but the only data we have has been shifted to prepare for the spline models.  So we unshift them.